jwt/jwt.go

```diff
@@ -161,31 +159,56 @@ func Verify(req *http.Request, keyServer keyserver.Reader, nonceVerifier noncest
 		return nil, err
 	} else if err != nil {
 		log.Errorf("Could not get public key from key server: %s", err)
-		return nil, errors.New("Unexpected key server error")
+		return nil, constructVerificationError("Unexpected key server error", protocol, publicBasePath, req)
 	}
 
 	verifier, err := publicKey.Verifier()
 	if err != nil {
 		log.Errorf("Could not create JWT verifier for public key '%s': %s", publicKey.ID(), err)
-		return nil, errors.New("Unexpected verifier initialization failure")
+		return nil, constructVerificationError("Unexpected verifier initialization failure", protocol, publicBasePath, req)
 	}
 
 	if verifier.Verify(jwt.Signature, []byte(jwt.Data())) != nil {
-		return nil, errors.New("Invalid JWT signature")
+		return nil, constructVerificationError("Invalid JWT signature", protocol, publicBasePath, req)
 	}
 
 	return claims, nil
 }
 
+func constructVerificationError(message string, protocol string, pathPrefix string, req *http.Request) error {
+	url := composeRedirectURL(protocol, req, pathPrefix)
```

Comment:
Does it make sense to send redirect if `cookiesEnabled` is `false`?
I mean will `loader.js` send token via HEADER or query parameter or it stores token in cookies only?

Reply:
Well, it depends.. All we know at this point in time is that the verification of the access token failed and that we have a redirect URL for performing the authentication. The JWT proxy cannot assume how that is done - `loader.js` might as well not use the `.../jwt/auth` endpoint at all and store the bearer token in some global variable in the webapp  and then set the token in each request in the header instead of `../jwt/auth` storing the token in the cookie.